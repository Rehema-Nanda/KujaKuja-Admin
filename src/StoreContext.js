// https://mobx-react.js.org/recipes-context

import React from "react";
import PropTypes from "prop-types";
import RootStore from "./stores/RootStore";

export const StoreContext = React.createContext(null);

export const StoreProvider = ({ children }) => {
    const store = new RootStore();
    return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};
StoreProvider.propTypes = {
    children: PropTypes.node.isRequired,
};

export const StoreConsumer = StoreContext.Consumer;

export const useStore = () => {
    const store = React.useContext(StoreContext);
    if (!store) {
        // this is especially useful in TypeScript so you don't need to be checking for null all the time
        throw new Error("useStore must be used within a StoreProvider.");
    }
    return store;
};
