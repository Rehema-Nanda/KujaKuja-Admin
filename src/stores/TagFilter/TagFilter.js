import {
    decorate, computed, comparer,
} from "mobx";

class TagFilter {
    store = null;
    id = null;
    tagText = null;
    searchText = null;
    startDate = null;
    endDate = null;
    createdAt = null;
    updatedAt = null;
    status = null;
    locations = null;

    constructor(store, json = null) {
        this.store = store;

        if (json) {
            this.id = json.id;
            this.updateFromJson(json);
        }
    }

    get asJson() {
        return {
            tag_text: this.tagText,
            search_text: this.searchText,
            start_date: this.startDate,
            end_date: this.endDate,
            created_at: this.createdAt,
            updated_at: this.updatedAt,
            status: this.status,
            locations: this.locations,
        };
    }

    updateFromJson(json) {
        this.tagText = json.tag_text;
        this.searchText = json.search_text;
        this.startDate = json.start_date;
        this.endDate = json.end_date;
        this.createdAt = json.created_at;
        this.updatedAt = json.updated_at;
        this.status = json.status;
        this.locations = json.locations;
    }

    async save() {
        const json = this.asJson;
        if (this.id) {
            await this.store.apiService.updateTagFilter(this.id, json);
        }
        else {
            const responseData = await this.store.apiService.createTagFilter(json);
            this.id = parseInt(responseData.id, 10);
            this.status = this.status || "EDITING";
            this.createdAt = this.createdAt || new Date();
            await this.store.createTagFilter(this);
        }
    }

    async delete() {
        if (!this.id) {
            return;
        }

        await this.store.apiService.deleteTagFilter(this.id);
        await this.store.removeTagFilter(this);
    }
}

decorate(TagFilter, {
    asJson: computed({ equals: comparer.shallow }),
});

export default TagFilter;
