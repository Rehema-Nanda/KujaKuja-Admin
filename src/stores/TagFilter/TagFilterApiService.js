import ApiService from "../ApiService";

export default class TagFilterApiService extends ApiService {
    fetchAllTagFilters = async () => {
        const tagFiltersResponse = await this.makeHttpCall("tags/tag_filters");
        return tagFiltersResponse.data.data; // TODO: change data structure returned by API
    };

    runAllTagFilters = async () => {
        const tagFilters = await this.makeHttpCall("tags/bulk_tag");
        return tagFilters.data.data;
    };

    runTagFilterById = async (id) => {
        const tagResponse = await this.makeHttpCall(`tags/bulk_tag/${id}`);
        return tagResponse.data.data;
    };

    updateTagFilter = async (id, tagJson) => {
        const config = { data: tagJson };
        const updateTagFilterResponse = await this.makeHttpCall(`tags/tag_filters/${id}`, "put", config);
        return updateTagFilterResponse.data;
    };

    createTagFilter = async (tagJson) => {
        const config = { data: tagJson };
        const createTagFilterResponse = await this.makeHttpCall(`tags/tag_filters`, "post", config);
        createTagFilterResponse.data.id = createTagFilterResponse.data.id.toString();
        return createTagFilterResponse.data;
    }

    deleteTagFilter = async (id) => {
        const deleteTagFilterResponse = await this.makeHttpCall(`tags/tag_filters/${id}`, "delete");
        deleteTagFilterResponse.data.id = deleteTagFilterResponse.data.id.toString();
        return deleteTagFilterResponse.data;
    };

    fetchTagFilterById = async (id) => {
        const tagResponse = await this.makeHttpCall(`tags/tag_filters/${id}`);
        return tagResponse.data.data; // TODO: change data structure returned by API
    };

    undoTagFilterBulkTagById = async (id) => {
        const tagResponse = await this.makeHttpCall(`tags/bulk_tag/undo/${id}`);
        return tagResponse.data.data;
    }
}
