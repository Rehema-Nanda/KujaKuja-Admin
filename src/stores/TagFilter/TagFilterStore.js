import {
    flow, decorate, observable, computed, onBecomeObserved,
} from "mobx";
import openSocket from "socket.io-client";
import TagFilter from "./TagFilter";
import TagFilterApiService from "./TagFilterApiService";
import AppConfig from "../../AppConfig";

class TagFilterStore {
    tagFilters = [];
    isLoading = false;

    constructor(rootStore) {
        this.rootStore = rootStore;
        this.apiService = new TagFilterApiService(AppConfig.API_URL);

        this.loadTagFilters();

        const socket = openSocket(AppConfig.ROOT_API_URL, { transports: ["polling"] });

        socket.on("connected", () => {
            socket.emit("listening for tag filter change", {});
        });

        const openStream = () => {
            socket.on("tagFilterTableChanged", () => {
                this.loadTagFilters();
            });
        };
        onBecomeObserved(this, "tagFilters", openStream);
    }

    get tagFiltersSortedByTagText() {
        return this.tagFilters.sort((a, b) => {
            if (a.tagText < b.tagText) {
                return -1;
            }
            else if (a.tagText > b.tagText) {
                return 1;
            }
            return 0;
        });
    }

    loadTagFilters = flow(function* () { // eslint-disable-line func-names
        this.isLoading = true;

        try {
            const tagFiltersJson = yield this.apiService.fetchAllTagFilters();

            this.tagFilters.replace(tagFiltersJson.map((tagFilterJson) => {
                return new TagFilter(
                    this,
                    tagFilterJson,
                );
            }));
            this.isLoading = false;
        }
        catch (error) {
            console.error(`TagFilterStore failed to load tags. ${error.message}`); // eslint-disable-line no-console
            this.isLoading = false;
        }
    });

    updateTagFromServer = flow(function* (id) { // eslint-disable-line func-names
        const tagFilterId = parseInt(id, 10);
        this.isLoading = true;
        const tagsJson = yield this.apiService.fetchTagFilterById(id);
        let tagFilter = this.tagFilters.find((t) => {
            return t.id === tagFilterId;
        });

        if (!tagsJson && tagFilter) {
            // tag no longer exists on the server, remove it locally
            yield this.removeTagFilter(tagFilter);
            tagFilter = null;
        }
        if (tagsJson) {
            if (tagFilter) {
                tagFilter.updateFromJson(tagsJson);
            }
            else {
                tagFilter = new TagFilter(this, tagsJson);
                this.tagFilters.push(tagFilter);
            }
        }
        this.isLoading = false;
        return tagFilter;
    });

    removeTagFilter = flow(function* (tagFilter) { // eslint-disable-line func-names, require-yield
        this.tagFilters.splice(this.tagFilters.indexOf(tagFilter), 1);
    });

    createTagFilter = flow(function* (tagFilter) { // eslint-disable-line func-names, require-yield
        this.tagFilters.push(tagFilter);
    });
}

decorate(TagFilterStore, {
    tagFilters: observable,
    tagFiltersSortedByTagText: computed,
    isLoading: observable,
});

export default TagFilterStore;
