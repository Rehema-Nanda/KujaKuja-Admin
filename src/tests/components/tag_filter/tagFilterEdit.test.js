import React from "react";
import { mount } from "enzyme";
import TagFilterEdit from "../../../components/tag_filters/edit";
import { StoreProvider } from "../../../StoreContext";
import TagFilterStore from "../../../stores/TagFilter/TagFilterStore";
import TagFilter from "../../../stores/TagFilter/TagFilter";

jest.mock("../../../stores/TagFilter/TagFilterStore");

const tagFilters = [
    new TagFilter(
        null,
        {
            id: "1",
            tag_text: "tagtest",
            search_text: "test|sample",
            start_date: "2020-01-01T00:00:00",
            end_date: "2020-02-01T00:00:00",
            created_at: "2020-01-05T12:13:14",
            locations: "Test Location",
        },
    ),
];

const mockTagFilterDelete = jest.fn();
tagFilters[0].delete = mockTagFilterDelete;

const mockUpdateTagFilterFromServer = jest.fn(() => {
    return tagFilters[0];
});

TagFilterStore.mockImplementation(() => {
    return {
        tagFilters: tagFilters,
        updateTagFromServer: mockUpdateTagFilterFromServer,
    };
});

describe("TagFilterEdit component", () => {
    const props = {
        history: {
            push: jest.fn(),
        },
        match: {
            params: {
                id: 1,
            },
        },
    };

    const wrapper = mount(<TagFilterEdit {...props} />, {
        wrappingComponent: StoreProvider,
    });

    it("should render correctly", () => {
        expect(mockUpdateTagFilterFromServer).toHaveBeenCalled();
        wrapper.update();
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should change modal state when toggleDeleteConfirmationModal function is called", () => {
        const modalState = wrapper.instance().simpleState.isDeleteConfirmationModalOpen;
        const toggleDeleteConfirmationModal = jest.spyOn(wrapper.instance(), "toggleDeleteConfirmationModal");
        toggleDeleteConfirmationModal();
        expect(wrapper.instance().simpleState.isDeleteConfirmationModalOpen).toEqual(!modalState);
    });

    it("should redirect to list page upon element delete", async () => {
        const deleteFn = jest.spyOn(wrapper.instance(), "delete");
        await deleteFn();
        expect(mockTagFilterDelete).toHaveBeenCalled();
        expect(props.history.push).toHaveBeenCalledWith("/tag_filters");
    });
});
