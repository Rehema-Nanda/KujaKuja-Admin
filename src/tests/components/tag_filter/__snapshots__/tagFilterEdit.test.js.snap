// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagFilterEdit component should render correctly 1`] = `
"<TagFilterEdit history={{...}} match={{...}}>
  <div className=\\"page\\">
    <div className=\\"title-background-parent\\">
      <div className=\\"title-background-shape\\" />
    </div>
    <Container tag=\\"div\\">
      <div className=\\"container\\">
        <Row tag=\\"div\\">
          <div className=\\"row\\">
            <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
              <div className=\\"col-sm-7 col-md-8\\">
                <div className=\\"page-title\\">
                  <h1 className=\\"left\\">
                    Tag Filters
                  </h1>
                  <p>
                    Manage your tag filter
                  </p>
                </div>
              </div>
            </Col>
            <Col sm={{...}} md={{...}} tag=\\"div\\" widths={{...}}>
              <div className=\\"col-sm-5 col-md-4\\">
                <div className=\\"bulk-update-button\\">
                  <Dropdown isOpen={false} toggle={[Function]} direction=\\"down\\" nav={false} active={false} addonType={false} inNavbar={false} setActiveFromChild={false}>
                    <Manager tag=\\"div\\" className=\\"dropdown\\" onKeyDown={[Function: bound handleKeyDown]}>
                      <div className=\\"dropdown\\" onKeyDown={[Function: bound handleKeyDown]}>
                        <DropdownToggle caret={true} color=\\"primary\\" className=\\"primary\\" aria-haspopup={true}>
                          <Target aria-haspopup={true} color=\\"primary\\" cssModule={[undefined]} className=\\"primary dropdown-toggle\\" component={[Function: Button]} onClick={[Function: bound onClick]} aria-expanded={false}>
                            <Button aria-haspopup={true} color=\\"primary\\" cssModule={[undefined]} className=\\"primary dropdown-toggle\\" onClick={[Function: bound onClick]} aria-expanded={false} innerRef={[Function: targetRef]} tag=\\"button\\">
                              <button type=\\"button\\" aria-haspopup={true} onClick={[Function: bound onClick]} aria-expanded={false} className=\\"primary dropdown-toggle btn btn-primary\\" aria-label={{...}}>
                                Actions
                              </button>
                            </Button>
                          </Target>
                        </DropdownToggle>
                        <DropdownMenu tag=\\"div\\" flip={true}>
                          <div tabIndex=\\"-1\\" role=\\"menu\\" aria-hidden={true} className=\\"dropdown-menu\\" x-placement={[undefined]}>
                            <DropdownItem className=\\"bulk-update-dropdown\\" onClick={[Function]} disabled={true} tag=\\"button\\" toggle={true}>
                              <button type=\\"button\\" onClick={[Function: bound onClick]} disabled={true} tabIndex=\\"-1\\" className=\\"bulk-update-dropdown disabled dropdown-item\\">
                                <span className=\\"bulk-update-dropdown-title\\">
                                  Delete filter
                                </span>
                              </button>
                            </DropdownItem>
                            <DropdownItem className=\\"bulk-update-dropdown\\" onClick={[Function: onClick]} disabled={true} tag=\\"button\\" toggle={true}>
                              <button type=\\"button\\" onClick={[Function: bound onClick]} disabled={true} tabIndex=\\"-1\\" className=\\"bulk-update-dropdown disabled dropdown-item\\">
                                <span className=\\"bulk-update-dropdown-title\\">
                                  Apply filter
                                </span>
                              </button>
                            </DropdownItem>
                            <DropdownItem className=\\"bulk-update-dropdown\\" onClick={[Function: onClick]} disabled={true} tag=\\"button\\" toggle={true}>
                              <button type=\\"button\\" onClick={[Function: bound onClick]} disabled={true} tabIndex=\\"-1\\" className=\\"bulk-update-dropdown disabled dropdown-item\\">
                                <span className=\\"bulk-update-dropdown-title\\">
                                  Undo filter
                                </span>
                              </button>
                            </DropdownItem>
                          </div>
                        </DropdownMenu>
                      </div>
                    </Manager>
                  </Dropdown>
                </div>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
    <ConfirmationModal buttonFunction={[Function]} toggle={[Function]} isOpen={false}>
      <Modal isOpen={false} toggle={[Function]} autoFocus={true} centered={false} role=\\"dialog\\" backdrop={true} keyboard={true} zIndex={1050} fade={true} onOpened={[Function: noop]} onClosed={[Function: noop]} modalTransition={{...}} backdropTransition={{...}} />
    </ConfirmationModal>
    <Container tag=\\"div\\">
      <div className=\\"container\\">
        <div className=\\"white-page-bg\\">
          <Row tag=\\"div\\">
            <div className=\\"row\\">
              <Col lg={{...}} tag=\\"div\\" widths={{...}}>
                <div className=\\"col-lg-10 offset-lg-1\\">
                  <div>
                    <AvForm onValidSubmit={[Function]} tag={[Function: Form]} model={{...}} validationEvent={{...}} method=\\"get\\" onSubmit={[Function: onSubmit]} onKeyDown={[Function: onKeyDown]} onInvalidSubmit={[Function: onInvalidSubmit]}>
                      <Form noValidate={true} action=\\"#\\" method=\\"get\\" onSubmit={[Function]} onKeyDown={[Function: onKeyDown]} className=\\"av-valid\\" tag=\\"form\\">
                        <form noValidate={true} action=\\"#\\" method=\\"get\\" onSubmit={[Function]} onKeyDown={[Function: onKeyDown]} className=\\"av-valid\\">
                          <Row tag=\\"div\\">
                            <div className=\\"row\\">
                              <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                <div className=\\"col-sm-6\\">
                                  <AvField name=\\"tag_text\\" label=\\"Tag Name\\" required={true} value=\\"tagtest\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"tag_text\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"tag_text\\" className=\\"\\">
                                              Tag Name
                                            </label>
                                          </Label>
                                          <AvInput id=\\"tag_text\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"tag_text\\" required={true} value=\\"tagtest\\" validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"tag_text\\" required={true} value=\\"tagtest\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"tag_text\\" type=\\"text\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"tag_text\\" required={true} value=\\"tagtest\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"tag_text\\" type=\\"text\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                </div>
                              </Col>
                              <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                <div className=\\"col-sm-6\\">
                                  <AvField disabled={true} name=\\"status\\" label=\\"Status\\" value={[undefined]}>
                                    <AvGroup check={false} disabled={true} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={true} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"status\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"status\\" className=\\"\\">
                                              Status
                                            </label>
                                          </Label>
                                          <AvInput id=\\"status\\" className={[undefined]} size={[undefined]} disabled={true} readOnly={[undefined]} name=\\"status\\" value={[undefined]} validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={true} readOnly={[undefined]} name=\\"status\\" value={[undefined]} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"status\\" type=\\"text\\">
                                              <input size={[undefined]} disabled={true} readOnly={[undefined]} name=\\"status\\" value={[undefined]} onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"status\\" type=\\"text\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                </div>
                              </Col>
                              <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                <div className=\\"col-sm-12\\">
                                  <AvField name=\\"search_text\\" label=\\"Search Text\\" value=\\"test|sample\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"search_text\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"search_text\\" className=\\"\\">
                                              Search Text
                                            </label>
                                          </Label>
                                          <AvInput id=\\"search_text\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"search_text\\" value=\\"test|sample\\" validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"search_text\\" value=\\"test|sample\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"search_text\\" type=\\"text\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"search_text\\" value=\\"test|sample\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"search_text\\" type=\\"text\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                </div>
                              </Col>
                              <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                <div className=\\"col-sm-10 offset-sm-2\\">
                                  <Table className=\\"search-examples\\" tag=\\"table\\" responsiveTag=\\"div\\">
                                    <table className=\\"search-examples table\\">
                                      <thead>
                                        <tr>
                                          <th colSpan=\\"4\\">
                                            Search text examples
                                          </th>
                                        </tr>
                                      </thead>
                                      <tbody>
                                        <tr>
                                          <td>
                                            <strong>
                                              food &amp; water
                                            </strong>
                                          </td>
                                          <td>
                                            <em>
                                              both &quot;food&quot; 
                                              <strong>
                                                and
                                              </strong>
                                               &quot;water&quot; are present in text
                                            </em>
                                          </td>
                                        </tr>
                                        <tr>
                                          <td>
                                            <strong>
                                              food | water
                                            </strong>
                                          </td>
                                          <td>
                                            <em>
                                              either &quot;food&quot; 
                                              <strong>
                                                or
                                              </strong>
                                               &quot;water&quot; are present in text
                                            </em>
                                          </td>
                                        </tr>
                                        <tr>
                                          <td>
                                            <strong>
                                              !food
                                            </strong>
                                          </td>
                                          <td>
                                            <em>
                                              &quot;food&quot; 
                                              <strong>
                                                not
                                              </strong>
                                               present in text
                                            </em>
                                          </td>
                                        </tr>
                                        <tr>
                                          <td>
                                            <strong>
                                              food &amp; (water | thirst) &amp; !electricity
                                            </strong>
                                          </td>
                                          <td>
                                            <em>
                                              &quot;food&quot; and either &quot;water&quot; or &quot;thirst&quot;, and not &quot;electricity&quot;, present in text
                                            </em>
                                          </td>
                                        </tr>
                                      </tbody>
                                    </table>
                                  </Table>
                                </div>
                              </Col>
                              <Col className=\\"dates-parent\\" sm={{...}} tag=\\"div\\" widths={{...}}>
                                <div className=\\"dates-parent col-sm-6\\">
                                  <AvField name=\\"start_date\\" id=\\"start_date\\" label=\\"Start Date (e.g 2020-03-31)\\" validate={{...}} placeholder=\\"YYYY-MM-DD\\" value=\\"2020-01-01\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"start_date\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"start_date\\" className=\\"\\">
                                              Start Date (e.g 2020-03-31)
                                            </label>
                                          </Label>
                                          <AvInput id=\\"start_date\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"start_date\\" validate={{...}} placeholder=\\"YYYY-MM-DD\\" value=\\"2020-01-01\\" validationEvent=\\"\\" trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"start_date\\" placeholder=\\"YYYY-MM-DD\\" value=\\"2020-01-01\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} pattern={{...}} className=\\"is-untouched is-pristine av-valid\\" id=\\"start_date\\" type=\\"text\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"start_date\\" placeholder=\\"YYYY-MM-DD\\" value=\\"2020-01-01\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} pattern={{...}} id=\\"start_date\\" type=\\"text\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                </div>
                              </Col>
                              <Col className=\\"dates-parent\\" sm={{...}} tag=\\"div\\" widths={{...}}>
                                <div className=\\"dates-parent col-sm-6\\">
                                  <AvField name=\\"end_date\\" id=\\"end_date\\" label=\\"End Date (e.g 2020-03-31)\\" validate={{...}} placeholder=\\"YYYY-MM-DD\\" value=\\"2020-02-01\\">
                                    <AvGroup check={false} disabled={[undefined]} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={[undefined]} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"end_date\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"end_date\\" className=\\"\\">
                                              End Date (e.g 2020-03-31)
                                            </label>
                                          </Label>
                                          <AvInput id=\\"end_date\\" className={[undefined]} size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"end_date\\" validate={{...}} placeholder=\\"YYYY-MM-DD\\" value=\\"2020-02-01\\" validationEvent=\\"\\" trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"end_date\\" placeholder=\\"YYYY-MM-DD\\" value=\\"2020-02-01\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} pattern={{...}} className=\\"is-untouched is-pristine av-valid\\" id=\\"end_date\\" type=\\"text\\">
                                              <input size={[undefined]} disabled={[undefined]} readOnly={[undefined]} name=\\"end_date\\" placeholder=\\"YYYY-MM-DD\\" value=\\"2020-02-01\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} pattern={{...}} id=\\"end_date\\" type=\\"text\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                </div>
                              </Col>
                              <Col className=\\"test-search-text-button-parent\\" sm={{...}} tag=\\"div\\" widths={{...}}>
                                <div className=\\"test-search-text-button-parent col-sm-12\\">
                                  <button type=\\"button\\" className=\\"test-search-text-button\\" onClick={[Function]}>
                                    Preview ideas
                                  </button>
                                </div>
                              </Col>
                              <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                <div className=\\"col-sm-12\\" />
                              </Col>
                              <Col sm={{...}} tag=\\"div\\" widths={{...}}>
                                <div className=\\"col-sm-12\\">
                                  <AvField disabled={true} name=\\"locations\\" label=\\"Locations\\" value=\\"Test Location\\">
                                    <AvGroup check={false} disabled={true} row={false}>
                                      <FormGroup className=\\"\\" check={false} disabled={true} row={false} tag=\\"div\\">
                                        <div className=\\"position-relative form-group\\">
                                          <Label for=\\"locations\\" className={[undefined]} hidden={[undefined]} size={[undefined]} tag=\\"label\\" widths={{...}}>
                                            <label htmlFor=\\"locations\\" className=\\"\\">
                                              Locations
                                            </label>
                                          </Label>
                                          <AvInput id=\\"locations\\" className={[undefined]} size={[undefined]} disabled={true} readOnly={[undefined]} name=\\"locations\\" value=\\"Test Location\\" validationEvent=\\"\\" validate={{...}} trueValue={true} falseValue={false} tag={[Function: Input]}>
                                            <Input size={[undefined]} disabled={true} readOnly={[undefined]} name=\\"locations\\" value=\\"Test Location\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} className=\\"is-untouched is-pristine av-valid\\" id=\\"locations\\" type=\\"text\\">
                                              <input size={[undefined]} disabled={true} readOnly={[undefined]} name=\\"locations\\" value=\\"Test Location\\" onKeyUp={[Function: bound onKeyUpHandler]} onBlur={[Function: bound onBlurHandler]} onInput={[Function: bound onInputHandler]} onFocus={[Function: bound onFocusHandler]} onChange={[Function: bound onChangeHandler]} id=\\"locations\\" type=\\"text\\" className=\\"is-untouched is-pristine av-valid form-control\\" />
                                            </Input>
                                          </AvInput>
                                        </div>
                                      </FormGroup>
                                    </AvGroup>
                                  </AvField>
                                </div>
                              </Col>
                            </div>
                          </Row>
                          <BtnRow buttonLabel=\\"Update\\" cancelForm={[Function]}>
                            <Row className=\\"btn-row\\" tag=\\"div\\">
                              <div className=\\"btn-row row\\">
                                <Col tag=\\"div\\" widths={{...}}>
                                  <div className=\\"col\\" />
                                </Col>
                                <Col tag=\\"div\\" widths={{...}}>
                                  <div className=\\"col\\">
                                    <Button color=\\"primary\\" className=\\"primary cancel\\" onClick={[Function]} tag=\\"button\\">
                                      <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"primary cancel btn btn-primary\\" aria-label={{...}}>
                                        Back
                                      </button>
                                    </Button>
                                  </div>
                                </Col>
                              </div>
                            </Row>
                          </BtnRow>
                        </form>
                      </Form>
                    </AvForm>
                  </div>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </div>
    </Container>
  </div>
</TagFilterEdit>"
`;
